drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1158 @@
+ * @param drv_cfg driver instance configuration
+ *
+ * @retval 0 on success
+ * @retval see @ref gpio_pin_set_dt
+ */
+static inline int tmag5273_dev_int_trigger(const struct tmag5273_config *drv_cfg)
+{
+	int retval;
+
+	retval = gpio_pin_configure_dt(&drv_cfg->int_gpio, GPIO_OUTPUT);
+	if (retval < 0) {
+		return retval;
+	}
+
+	retval = gpio_pin_set_dt(&drv_cfg->int_gpio, 1);
+	if (retval < 0) {
+		return retval;
+	}
+
+	retval = gpio_pin_set_dt(&drv_cfg->int_gpio, 0);
+	if (retval < 0) {
+		return retval;
+	}
+
+	retval = gpio_pin_configure_dt(&drv_cfg->int_gpio, GPIO_INPUT);
+	if (retval < 0) {
+		return retval;
+	}
+
+	return 0;
+}
+
+/** @brief returns the high measurement range based on the chip version */
+static inline uint16_t tmag5273_range_high(uint8_t version)
+{
+	return (version == TMAG5273_VER_TMAG5273X1) ? TMAG5273_MEAS_RANGE_HIGH_MT_VER1
+						    : TMAG5273_MEAS_RANGE_HIGH_MT_VER2;
+}
+
+/** @brief returns the low measurement range based on the chip version */
+static inline uint16_t tmag5273_range_low(uint8_t version)
+{
+	return (version == TMAG5273_VER_TMAG5273X1) ? TMAG5273_MEAS_RANGE_LOW_MT_VER1
+						    : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+}
+
+/**
+ * @brief update the measurement range of the X/Y/Z-axis
+ *
+ * @param dev handle to the sensor
+ * @param val value to be set
+ *
+ * @return see @ref i2c_reg_update_byte_dt
+ */
+static inline int tmag5273_attr_set_xyz_meas_range(const struct device *dev,
+						   const struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	const uint16_t range_high = tmag5273_range_high(drv_data->version);
+	const uint16_t range_low = tmag5273_range_low(drv_data->version);
+
+	int retval;
+	uint8_t regdata;
+	uint16_t range;
+
+	if (val->val1 >= range_high) {
+		regdata = TMAG5273_XYZ_MEAS_RANGE_HIGH;
+		range = range_high;
+	} else {
+		regdata = TMAG5273_XYZ_MEAS_RANGE_LOW;
+		range = range_low;
+	}
+
+	retval = i2c_reg_update_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2,
+					TMAG5273_MEAS_RANGE_XYZ_MSK, regdata);
+	if (retval < 0) {
+		return retval;
+	}
+
+	drv_data->xyz_range = range;
+
+	return 0;
+}
+
+/**
+ * @brief returns the used measurement range of the X/Y/Z-axis
+ *
+ * @param dev handle to the sensor
+ * @param val return value
+ *
+ * @return \c 0 on success
+ * @return see @ref i2c_reg_read_byte_dt
+ */
+static inline int tmag5273_attr_get_xyz_meas_range(const struct device *dev,
+						   struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	uint8_t regdata;
+	int retval;
+
+	retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, &regdata);
+	if (retval < 0) {
+		return retval;
+	}
+
+	if ((regdata & TMAG5273_MEAS_RANGE_XYZ_MSK) == TMAG5273_XYZ_MEAS_RANGE_HIGH) {
+		val->val1 = tmag5273_range_high(drv_data->version);
+	} else {
+		val->val1 = tmag5273_range_low(drv_data->version);
```

Comment:
I think this could use a `val->val2 = 0`

Reply:
Oh, missed this, thx